compiler/rustc_ty_utils/src/assoc.rs

```diff
@@ -21,9 +22,34 @@ pub fn provide(providers: &mut ty::query::Providers) {
 fn associated_item_def_ids(tcx: TyCtxt<'_>, def_id: DefId) -> &[DefId] {
     let item = tcx.hir().expect_item(def_id.expect_local());
     match item.kind {
-        hir::ItemKind::Trait(.., ref trait_item_refs) => tcx.arena.alloc_from_iter(
-            trait_item_refs.iter().map(|trait_item_ref| trait_item_ref.id.owner_id.to_def_id()),
-        ),
+        hir::ItemKind::Trait(.., ref trait_item_refs) => {
+            if tcx.sess.opts.unstable_opts.new_rpitit {
+                tcx.arena.alloc_from_iter(
```

Comment:
This could use a block comment explaining what the strategy here is. Basically that we collect the RPITITs from the method's return type and turn them into items (but with more words).

Reply:
Done